internal/cluster/validator.go

```diff
@@ -388,6 +388,12 @@ func (v *clusterValidator) SufficientMastersCount(c *clusterPreprocessContext) (
 		status = ValidationFailure
 	}
 
+	if numMasters < common.MinMasterHostsNeededForInstallationInHaMode &&
+		numMasters >= common.MinMasterHostsNeededForInstallationInHaArbiterMode &&
+		!common.IsClusterTopologyHighlyAvailableArbiter(c.cluster) {
```

Comment:
This is a strange condition. It's really saying if the number of masters is 2 and we don't have an arbiter then fail.

I feel like we could say that more directly. Maybe change `common.GetHostsByEachRole` to also support arbiter?